papis/commands/update.py

```diff
@@ -39,25 +120,223 @@
     :prog: papis update
 """
 
-from typing import List, Dict, Tuple, Optional, Any
+import ast
+from typing import Any, Dict, List, Optional, Tuple
 
 import click
 
-import papis.utils
-import papis.strings
+import papis.cli
+import papis.commands.doctor
+import papis.config
 import papis.document
 import papis.format
-import papis.cli
-import papis.importer
 import papis.git
+import papis.importer
 import papis.logging
+import papis.strings
+import papis.utils
 
 logger = papis.logging.get_logger(__name__)
 
-
-def run(document: papis.document.Document,
-        data: Optional[Dict[str, Any]] = None,
-        git: bool = False) -> None:
+KEY_TYPES = papis.commands.doctor.get_key_type_check_keys()
+
+
+def try_parsing_str(key: str, value: str) -> str:
+    """
+    Tries to parse the input string as a python expression.
+
+    :returns: The parsed input string if string is a python expression,
+        otherwise an unchanged string.
+    """
+    try:
+        value = ast.literal_eval(value)
+    except (SyntaxError, ValueError):
+        logger.debug("Value '%s' of key '%s' is not a python expression.", value, key)
+    return value
+
+
+def run_set(
+    document: papis.document.DocumentLike, to_set: List[Tuple[str, str]]
+) -> None:
```

Comment:
All the new functions seem to use the `black` formatting like here. It's probably fine to leave it as is, but we should see if we want to use a formatter to be more consistent?

Reply:
I'm mostly using ruff in my projects too these days, so definitely a fan! (Haven't tried the formatter yet)